### DISCUSSION TITLE
How to Put Entire HL7 Segments to the Channel Map?

### DISCUSSION THREAD

**Discussion by flugruger** [nextgenhealthcare/connect#5061] [Q&A] [answered]
To address some issues, I'd like to drop entire HL7 message segments to the channel map. 
What's the cleanest way to do that?

Edit: Found the solution... sharing here:

```
// Log Key HL7 Segments
var segmentMSH = SerializerFactory.getSerializer('HL7V2').fromXML(msg.MSH.copy())+ '\r';
channelMap.put('segmentMSH', segmentMSH);

var segmentPID = SerializerFactory.getSerializer('HL7V2').fromXML(msg.PID.copy())+ '\r';
channelMap.put('segmentPID', segmentPID);

var segmentPV1 = SerializerFactory.getSerializer('HL7V2').fromXML(msg.PV1.copy())+ '\r';
channelMap.put('segmentPV1', segmentPV1);

var segmentORC = SerializerFactory.getSerializer('HL7V2').fromXML(msg.ORC.copy())+ '\r';
channelMap.put('segmentORC', segmentORC);

var segmentOBR = SerializerFactory.getSerializer('HL7V2').fromXML(msg.OBR.copy())+ '\r';
channelMap.put('segmentOBR', segmentOBR);
```

**Comment by pacmano1** [comment, marked as answer]
I'd probably share the actual issues, no?

**Reply by flugruger** [reply]
Sure... @pacmano1 ...  it goes back to this build not handling base64 PDFs as attachments. Changing that would be significant. 

In the meantime, I'd like to increase the pruning but still have the HL7 available after pruning....  so in Metadata. 
My thought is to drop the MSH, PID, PV1, ORU and ORM segments into distinct channel map items. 

Thoughts? Suggestions?

**Reply by flugruger** [reply]
Found the solution:
```
// Log Key HL7 Segments
var segmentMSH = SerializerFactory.getSerializer('HL7V2').fromXML(msg.MSH.copy())+ '\r';
channelMap.put('segmentMSH', segmentMSH);

var segmentPID = SerializerFactory.getSerializer('HL7V2').fromXML(msg.PID.copy())+ '\r';
channelMap.put('segmentPID', segmentPID);

var segmentPV1 = SerializerFactory.getSerializer('HL7V2').fromXML(msg.PV1.copy())+ '\r';
channelMap.put('segmentPV1', segmentPV1);

var segmentORC = SerializerFactory.getSerializer('HL7V2').fromXML(msg.ORC.copy())+ '\r';
channelMap.put('segmentORC', segmentORC);

var segmentOBR = SerializerFactory.getSerializer('HL7V2').fromXML(msg.OBR.copy())+ '\r';
channelMap.put('segmentOBR', segmentOBR);
```

**Reply by pacmano1** [reply]
If you really don't want to persist parts of the message, just delete (or empty) the segments you don't want after doing something with them.    It's sort of an odd situation you have, it would be good for you to post a bit more on what the channel is doing, converting to an attachment based channel is usually not difficult.  For example you can collect the PDF to an attachment right on the inbound side when you receive the HL7 message.
